### PR TITLE
feat: implement dashboard

### LINKED ISSUE #39: feat: implement dashboard
We need a dashboard / explorer to view current and past tasks made by the Operators.

### PR DESCRIPTION
**Pull request by uri-99**
# Objective

This PR closes #39, implementing a simple dashboard / explorer to navigate and visualize tasks by their id; their content and status.

The explorer MVP seeks to:

* Search tasks
* Preview simple task data from the blockchain

### Set up environment variables

Run the following command to set up the environment variables:

```sh
make create_env
```

This will create a `.env` file in the `/explorer` directory of the project. The `.env` file will contain the following variables:

| Variable | Description |
| -------- | ----------- |
| `RPC_URL` | The RPC URL of the network you want to connect to. |
| `ENVIRONMENT` | The environment you want to run the application in. It can be `devnet`, `holesky` or `mainnet`. |

### Run devnet environment (optional)

If you want to run the devnet environment, you can run the following command in another terminal:

```sh
cd ..
make anvil-start
```

Then in another terminal, you can run the following command to run the operator:

```sh
make operator-full-registration
make operator-start
```

Then, in another terminal, run the following command to start the aggregator:

```sh
make aggregator-start
```

Finally, to have a task running in the devnet, you can run the following command:

```sh
make send-plonk_bls12_381-proof-loop
```

### Run the frontend

To start your Phoenix server:

```makefile
cd explorer
make <run | deps | help>
```

| Command | Description |
| --- | --- |
| `make run` | Starts the Elixir backend server. |
| `make deps` | Installs Elixir code dependencies. |
| `make help` | Shows the help message. |

Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
You can access to a tasks information by visiting `localhost:4000/tasks/:id`.

## Views

image

image

image

### CODE REVIEW

File: explorer/lib/explorer/application.ex [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  @impl true
+  def config_change(changed, _new, removed) do
+    ExplorerWeb.Endpoint.config_change(changed, removed)
+    :ok
+  end
+end
+
+defmodule MyERC20Token do
+  use Ethers.Contract,
+    abi_file: "lib/abi/UriCoin.json",
```
Comment (entropidelic) [inline explorer/lib/explorer/application.ex]: good coin, but we should probably use another ERC20

File: explorer/lib/explorer_web/controllers/task_controller.ex [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    # Returns the AlignedLayer "meaning" value
+    # data = AlignedLayerServiceManager.get_meaning() |> Ethers.call() |> IO.puts()
+
+    # Returns AlignedLayer latestTaskNum
+    # data = AlignedLayerServiceManager.latest_task_num() |> Ethers.call() |> IO.puts()
+
+    # Returns AlignedLayer is_aggregator -> bool
+    # data = AlignedLayerServiceManager.is_aggregator("0x703E7dE5F528fA828f3BE726802B2092Ae7deb2F") |> Ethers.call()
+
```
Comment (entropidelic) [inline explorer/lib/explorer_web/controllers/task_controller.ex]: could this be removed?

File: explorer/lib/abi/UriCoin.json [hunk 3]
```diff
@@ -0,0 +1 @@
...me":"from","type":"address","indexed":true,"internalType":"address"},{"name":"to","type":"address","indexed":true,"internalType":"address"},{"name":"value","type":"uint256","indexed":false,"internalType":"uint256"}],"anonymous":false},{"type":"error","name":"ERC20InsufficientAllowance","inputs":[{"name":"spender","type":"address","internalType":"address"},{"name":"allowance","type":"uint256","internalType":"uint256"},{"name":"needed","type":"uint256","internalType":"uint256"}]},{"type":"error","name":"ERC20InsufficientBalance","inputs":[{"name":"sender","type":"address","internalType":"address"},{"name":"balance","type":"uint256","internalType":"uint256"},{"name":"needed","type":"uint256","internalType":"uint256"}]},{"type":"error","name":"ERC20InvalidApprover","inputs":[{"name":"approver","type":"address","internalType":"address"}]},{"type":"error","name":"ERC20InvalidReceiver","inputs":[{"name":"receiver","type":"address","internalType":"address"}]},{"type":"error","name":"ERC20InvalidSender","inputs":[{"name":"sender","type":"address","internalType":"address"}]},{"type":"error","name":"ERC20InvalidSpender","inputs":[{"name":"spender","type":"address","internalType":"address"}]}]
```
Comment (entropidelic) [inline explorer/lib/abi/UriCoin.json]: uricoin to the moon 🚀

File: explorer/lib/explorer/application.ex (hunk 1 above)
Comment (uri-99) [reply explorer/lib/explorer/application.ex]: removed 👍

File: explorer/lib/explorer_web/controllers/task_controller.ex (hunk 2 above)
Comment (uri-99) [reply explorer/lib/explorer_web/controllers/task_controller.ex]: I removed the first one (get_erc20_name), but the rest of the gets will be used in the next explorer iteration (for example get_latest_task_num will be used to show last task)

File: explorer/lib/abi/UriCoin.json (hunk 3 above)
Comment (uri-99) [reply explorer/lib/abi/UriCoin.json]: removed 👍🚀

File: explorer/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+make send-plonk_bls12_381-proof-loop
+```
+
+### Run the frontend
+
+Set up your environment variables:
+
+```sh
+export RPC_URL=your_rpc_url
+```
```
Comment (entropidelic) [inline explorer/README.md]: is this step necessary now that we have the `.env` file?

**Review by entropidelic — APPROVED** [review]

File: explorer/README.md (hunk 4 above)
Comment (glpecile) [reply explorer/README.md]: Nope, its now included in the env

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**